app/components/avo/items/switcher_component.rb

```diff
@@ -53,12 +53,29 @@ def in_reflection?
   end
 
   def tab_group_component
-    Avo::TabGroupComponent.new resource: @resource, group: item.hydrate(view: view), index: index, params: params, form: form, view: view
+    Avo::TabGroupComponent.new(
+      resource: @resource,
+      group: item.hydrate(view: view),
+      index: index, params: params,
+      form: form,
+      view: view
+    )
   end
 
   def field_component
-    final_item = item.dup.hydrate(resource: @resource, record: @resource.record, user: resource.user, view: view)
-    final_item.component_for_view(@view).new(field: final_item, resource: @resource, index: index, form: form, turbo_frame_loading: :lazy, **@field_component_extra_args)
+    item.hydrate(
+      resource: @resource,
+      record: @resource.record,
+      user: resource.user,
+      view: view,
+    ).component_for_view(@view).new(
+      field: item,
```

Comment:
again, in the past code we were sending the `dup` item back.

Reply:
Old code:
```ruby
final_item = item.dup.hydrate(resource: @resource, record: @resource.record, user: resource.user, view: view)
final_item.component_for_view(@view).new(field: final_item, resource: @resource, index: index, form: form, turbo_frame_loading: :lazy, **@field_component_extra_args)
```

The "old" code was sending the `component_for_view` new instance back.

This would be useful if we want to keep 1 "dehydrated" item (called `item`) and 1 hydrated item (called `final_item`)

Maybe when this was implement there were a necessity to keep 1 "dehydrated" item, right now, from my analysis there is no need to keep the `item` "dehydrated"  and it may have some little impact on the performance since each field instance is getting duplicated.

Reply:
what I meant was that the `dup`ing would ensure that any changes that would be applied to the `dup`ed object would not seep into the original one that was being used somewhere else.
It's a different object in memory.

Reply:
Reverted it just in case